include/circt/Dialect/Pipeline/PipelineOps.td

```diff
@@ -169,27 +197,29 @@ def ScheduledPipelineOp : Op<Pipeline_Dialect, "scheduled", [
     // must be defined within a stage".
     bool isMaterialized();
 
-    // Returns the arguments for a stage. The stage valid signal is _not_ part
+    // Returns the data arguments for a stage. The stage enable signal is _not_ part
     // of the returned values.
-    ValueRange getStageArguments(Block* stage) {
-      if(stage != getEntryStage())
-        return stage->getArguments().drop_back();
+    llvm::ArrayRef<BlockArgument> getStageDataArgs(Block* stage) {
+      if(stage == getEntryStage())
+        return getInnerInputs();
 
-      // The entry stage block mirrors the input argument order of the pipeline.
-      // Grab the arguments corresponding to the inputs.
-      return stage->getArguments().slice(getInputs().getBeginOperandIndex(), getInputs().size());
+      // Data arguments for all non-entry stages are all block arguments
+      // except for the last block arguments (which is the stage enable signal).
+      return stage->getArguments().drop_back();
     }
 
-    // Returns the valid signal for the given stage. The valid signal is always
-    // the last signal in the stage argument list.
-    Value getStageValidSignal(Block* stage) {
-      return stage->getArguments().back();
+    // Returns the enable signal of the given pipeline stage. This is always
+    // the last block argument of a stage for anything but the entry stage.
+    Value getStageEnableSignal(size_t stageIdx) {
```

Comment:
This is an attempt to make things a bit more consistent wrt:
* The `enable` signal refers to the **input** that a stage receives
* The `valid` signal refers to the **output** that a stage sends to the next stage

Confusion (prior and now) is thus easy to find, given that each stage connects its enable -> valid internally (since stages have single cycle latency, for now...).